### ISSUE TITLE
Chunks at the edge of view distance suddenly pop into existance

### ISSUE THREAD

**Issue by TrollyLoki** [Minestom/Minestom#2455]
Minestom loads chunks at a slightly different distance from vanilla causing the fog to not be effective in hiding chunk loading.
I think this is due to the fact that vanilla adds an additional chunk to the actual view distance when deciding how many chunks to load.

See examples below comparing a vanilla server and a minestom server with equivalent view distance (8) and the same client.

## Vanilla:
![vanilla](https://github.com/user-attachments/assets/40e714cf-a27c-4f82-a5c8-27f1a1bfe3d5)

## Minestom:
![minestom](https://github.com/user-attachments/assets/af4f835c-db61-4c05-a055-8391b914939e)

**Comment by Nex0662** [comment]
You are right that with vanilla, the client (and server) tends to load additional chunks beyond the view distance, which is where you see the fog animation during the chunk loading, where as Minestom tends to stick strictly to the defined view distance without loading those additional chunks, which results in the effect you've shown at the bottom video. 

Something that you could do to go around this is by modifying the chunk loading behaviour to include additional layers of the chunks beyond the render distance, that might be a possible workaround to this, if that helps at least :)
